src/Server/MySQLHandler.cpp

```diff
@@ -328,6 +328,9 @@ void MySQLHandler::comQuery(ReadBuffer & payload)
 
         Context query_context = connection_context;
 
+        size_t affected_rows = 0;
+        query_context.setProgressCallback([&](const Progress & progress) { affected_rows += progress.written_rows; });
```

Comment:
Is it possible that `connection_context` already have some progress callback? Maybe do something similar to [this](https://github.com/ClickHouse/ClickHouse/blob/55d05c95bfcd43c9b094da5ff389d7be1f0ad365/src/Server/HTTPHandler.cpp#L584)?